evidence/create_build.go

```diff
@@ -0,0 +1,128 @@
+package evidence
+
+import (
+	"fmt"
+	"github.com/jfrog/jfrog-cli-artifactory/evidence/utils"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
+	coreConfig "github.com/jfrog/jfrog-cli-core/v2/utils/config"
+	"github.com/jfrog/jfrog-client-go/artifactory"
+	"github.com/jfrog/jfrog-client-go/artifactory/services"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"strings"
+)
+
+type createEvidenceBuild struct {
+	createEvidenceBase
+	project string
+	build   string
+}
+
+func NewCreateEvidenceBuild(serverDetails *coreConfig.ServerDetails, predicateFilePath string, predicateType string, key string, keyId string,
+	project string, build string) Command {
+	return &createEvidenceBuild{
+		createEvidenceBase: createEvidenceBase{
+			serverDetails:     serverDetails,
+			predicateFilePath: predicateFilePath,
+			predicateType:     predicateType,
+			key:               key,
+			keyId:             keyId,
+		},
+		project: project,
+		build:   build,
+	}
+}
+
+func (c *createEvidenceBuild) CommandName() string {
+	return "create-build-evidence"
+}
+
+func (c *createEvidenceBuild) ServerDetails() (*config.ServerDetails, error) {
+	return c.serverDetails, nil
+}
+
+func (c *createEvidenceBuild) Run() error {
+	artifactoryClient, err := c.createArtifactoryClient()
+	if err != nil {
+		log.Error("failed to create Artifactory client", err)
+		return err
+	}
+	subject, err := c.buildBuildInfoSubjectPath(artifactoryClient)
+	if err != nil {
+		return err
+	}
+	envelope, err := c.createEnvelope(subject)
+	if err != nil {
+		return err
+	}
+	err = c.uploadEvidence(envelope, subject)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *createEvidenceBuild) buildBuildInfoSubjectPath(artifactoryClient artifactory.ArtifactoryServicesManager) (string, error) {
+	build := strings.Split(c.build, ":")
+	if len(build) != 2 {
+		return "", fmt.Errorf("invalid build format. expected format is <name>:<number>")
+	}
+	name := build[0]
+	number := build[1]
+
+	timestamp, err := getBuildLatestTimestamp(name, number, c.project, artifactoryClient)
+	if err != nil {
+		return "", err
+	}
+
+	repoKey := buildBuildInfoRepoKey(c.project)
+	buildInfoPath := buildBuildInfoPath(repoKey, name, number, timestamp)
+	buildInfoChecksum, err := getBuildInfoPathChecksum(buildInfoPath, artifactoryClient)
+	if err != nil {
+		return "", err
+	}
+	return buildInfoPath + "@" + buildInfoChecksum, nil
+}
+
+func getBuildLatestTimestamp(name string, number string, project string, artifactoryClient artifactory.ArtifactoryServicesManager) (string, error) {
+	buildInfo := services.BuildInfoParams{
+		BuildName:   name,
+		BuildNumber: number,
+		ProjectKey:  project,
+	}
+	res, b, err := artifactoryClient.GetBuildInfo(buildInfo)
+	if !b {
+		errorMessage := fmt.Sprintf("failed to find build, name:%s, number:%s, project: %s", name, number, project)
+		if err != nil {
```

Comment:
The order of conditions looks wrong, the `err` should be checked first:
```go
if err != nil {
    return "", err
}
if !ok {
    errorMessage := fmt.Sprintf("failed to find build, name:%s, number:%s, project: %s", name, number, project)
    return "", errorutils.CheckErrorf(errorMessage)
}
```

Reply:
done